packages/bot/src/stores/toolbar-store.js

```diff
@@ -0,0 +1,283 @@
+
+    @action.bound onLoadClick = ({ is_local }) =>  {
+        if (is_local) {
+            const upload = document.getElementById('files');
+            upload.click();
+        } else {
+            // TO DO
+        }
+    }
+
+    @action.bound handleFileChange = e => {
+        let files,
+            dropEvent;
+        if (e.type === 'drop') {
+            e.stopPropagation();
+            e.preventDefault();
+            ({ files } = e.dataTransfer);
+            dropEvent = e;
+        } else {
+            ({ files } = e.target);
+
+            this.toggleSaveLoadModal();
+        }
+        files = Array.from(files);
+        files.forEach(file => {
+            if (file.type.match('text/xml')) {
+                this.readFile(file, dropEvent);
+            } else {
+                globalObserver.emit('ui.log.info', `${translate('File is not supported:')} ${file.name}`);
+            }
+        });
+        e.target.value = '';
+    }
+
+    @action.bound onConfirmSave = ({ is_local, save_as_collection }) => {
+        if (is_local) {
+            const file_name = this.file_name;
+            const xml = Blockly.Xml.workspaceToDom(Blockly.derivWorkspace);
+            xml.setAttribute('collection', save_as_collection ? 'true' : 'false');
+
+            const data = Blockly.Xml.domToPrettyText(xml);
+            const blob = new Blob([data], { type: 'text/xml;charset=utf-8' });
+
+            filesaver.saveAs(blob, file_name);
+        } else {
+            // TO DO
+        }
+
+        this.toggleSaveLoadModal();
+    }
+
+    @action.bound onUndoClick = () => {
+        Blockly.Events.setGroup('undo');
+        Blockly.derivWorkspace.undo();
+        Blockly.Events.setGroup(false);
+    }
+
+    @action.bound onRedoClick = () => {
+        Blockly.derivWorkspace.undo(true);
+    }
+
+    @action.bound onZoomInOutClick = is_zoom_in => {
+        const metrics = Blockly.derivWorkspace.getMetrics();
+        if (is_zoom_in) {
+            Blockly.derivWorkspace.zoom(metrics.viewWidth / 2, metrics.viewHeight / 2, 1);
+        } else {
+            Blockly.derivWorkspace.zoom(metrics.viewWidth / 2, metrics.viewHeight / 2, -1);
+        }
+    }
+
+    @action.bound onSortClick = () => {
+        Blockly.Events.setGroup(true);
+        const topBlocks = Blockly.derivWorkspace.getTopBlocks(true);
+        let cursorY = 0;
+        topBlocks.forEach(block => {
+            if (block.getSvgRoot().style.display !== 'none') {
+                const xy = block.getRelativeToSurfaceXY();
+                block.moveBy(-xy.x, cursorY - xy.y);
+                block.snapToGrid();
+                cursorY =
+                    block.getRelativeToSurfaceXY().y + block.getHeightWidth().height + Blockly.BlockSvg.MIN_BLOCK_Y;
+            }
+        });
+        Blockly.Events.setGroup(false);
+        // Fire an event to allow scrollbars to resize.
+        Blockly.derivWorkspace.resizeContents();
+    }
+
+    @computed
+    get isModalOpen() {
```

Comment:
Do we need getters for this, can we not just access the observable value itself? `@computed` should be used for computed values, e.g. 

```JavaScript
get myFunction() {
    return this.a * this.b;
}
```